kivy/uix/textinput.py

```diff
@@ -1574,22 +1600,42 @@ def on_touch_up(self, touch):
         touch.ungrab(self)
         self._touch_count -= 1
 
-        # schedule long touch for paste
-        if self._long_touch_ev is not None:
-            self._long_touch_ev.cancel()
-            self._long_touch_ev = None
+        self.cancel_long_touch_event()
 
         if not self.focus:
             return False
 
+        _scroll_timeout = touch.time_update - touch.time_start
+        any_condition_true = any((
```

Comment:
there are not very descriptive names, and there is no comment to explain the logic, i don't think extracting them from the conditions clarifies it much, also although it's a minor point, using any/all here is less efficient than using `and` or `or`, as it'll need to evaluate each expression before passing them to the function call, instead of stopping at the first value allowing to determine the expression (in an `and` the first falsy value, in an `any` the first truthy one), and function calls in python are not free, the performances are not a big deal there, but i find it less idiomatic than just using the python keywords.

Reply:
I agree with you, I believe I focused more on making the feature work than on the aspects you mentioned.  😂
I will modify